tensorflow_datasets/core/dataset_builder.py

```diff
@@ -287,7 +287,7 @@ def download_and_prepare(self, download_dir=None, download_config=None):
       return
 
     if self.version.tfds_version_to_prepare:
-      available_to_prepare = ", ".join(str(v) for v in self.versions
+      available_to_prepare = ", ".join(str(v) for v in [self.canonical_version] + [max(self.versions)]
```

Comment:
You should add a test. This won't raise an error in most cases as  most dataset will ever go in the `if self.version.tfds_version_to_prepare:` scope

Reply:
Ok so test should be in [this](https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/core/dataset_builder_test.py#L107) class or I have to create class for `dataset_builder.download_and_prepare` ?